src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -273,6 +310,22 @@ def reduce(
             tensor = sync_ddp_if_available(tensor, group, reduce_op=reduce_op)
         return tensor
 
+    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:
```

Comment:
Why aren't we using the precision context here? Please add a comment.

Reply:
because casting is done by FSDP. add a comment.